packages/mask/src/plugins/Avatar/SNSAdaptor/NFTBadge.tsx

```diff
@@ -33,8 +33,12 @@ function formatPrice(amount: string, symbol: string) {
     return `${_amount.div(1e6).toFixed(1)}M ${symbol}`
 }
 
-function formatText(symbol: string, length: number) {
-    return symbol.length > length ? `${symbol.slice(0, length)}...` : symbol
+function formatText(name: string, tokenId: string) {
+    const _name = name.replace(/#\d*/, '')
+    if (tokenId.length > 10) {
+        return `${_name.trim()} #${tokenId.substr(0, 6)}...${tokenId.substr(-4)}`
+    }
+    return `${_name} #${tokenId}`
```

Comment:
```suggestion
    name = name.replace(/#\d*/, '').trim()
    tokenId = wrap(token)
    if (tokenId.length > 10) {
        token = tokenId.slice(0, 6) + '...' + tokenId.slice(-4)
    }
    return `${name} #${tokenId}`
```

plz DO NOT USE `substr` and `strstring`, thanks